[Merl 718] Add Getting started guide for wp-graphql-conten-blocks plugin (#1252)
wpengine/faustjs: internal/faustjs.org/docs/gutenberg/wp-graphql-content-blocks.mdx

```graphql
contentBlocks {
    __typename
    name
    ...on CoreColumns {
		innerBlocks {
			...on CoreColumn {
				...on CoreParagraph {
					attributes {
						content
						className
					}
				}
			}
		}
    }
}
```

This clearly becomes an scalability issue as we don't really know how many levels deep can we use for those types. It could be two levels or it could be 50. Using [block patterns](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-patterns/) for example posses a real issue.
In each case we would have to copy the whole fragment list again and again. This is one of the shorfalls of `wp-graphql-gutenberg` plugin but with this plugin we have a more elegant solution.

This is how you request all blocks as a flat list:

```graphql
contentBlocks(flat:true) {
    __typename
    name
	id: nodeId
	parentId
    ... on CoreColumns {
        attributes {
        	className
        }
    }
	... on CoreColumn {
        attributes {
        	className
        }
    }
	...on CoreParagraph {
		attributes {
			content
			className
		}
	}
}
```

Given the flattened list of blocks though, how can you put it back? Well that's where you use the

> `nodeId` and `parentId` fields to assign temporal unique ids for each block.

Reviewer: ```suggestion
`nodeId` and `parentId` fields to assign temporary unique ids for each block.
```
Did you mean 'temporary' here? If you meant 'temporal' it would help to explain it- I tried googling and didn't see anything pop out at me.

> The `nodeId` field assignes a temporal unique id for a specific block and the `parentId` will

Reviewer: ```suggestion
The `nodeId` field assigns a temporary unique id for a specific block and the `parentId` will
```
I have the same question about 'temporal' or 'temporary' here too as above.

> be assigned only if the current block has a parent and it will get it's `nodeId` value.

Reviewer: ```suggestion
be assigned only if the current block has a parent. If the current block does have a parent, it will get the parent's `nodeId` value.
```

So in order to put everything back in the Headless site, you want to use the `flatListToHierarchical` function as mentioned in the [WPGraphQL docs](https://www.wpgraphql.com/docs/menus#hierarchical-data).

> This way you can reconstruct the block tree as before and pass it on the `WordPressBlocksViewer` component:

Reviewer: ```suggestion
This way you can reconstruct the block tree as before and pass it on to the `WordPressBlocksViewer` component:
```

```js
import { WordPressBlocksViewer } from '@faustwp/blocks';
import flatListToHierarchical from '../utilities/flatListToHierarchical';
...
const { contentBlocks } = props.data.post;
const blocks = flatListToHierarchical(contentBlocks);

<WordPressBlocksViewer contentBlocks={blocks} />
...
```
:::note

> Currently the `nodeId` field is only unique per request and it is not persisted anywhere. If you perform another request each block will be assigned a new `nodeId` each time.

Reviewer: ```suggestion
Currently the `nodeId` field is only unique per request and is not persisted anywhere. If you perform another request each block will be assigned a new `nodeId` each time.
```
